### DISCUSSION TITLE
how you can use 5 letter material unicode with windows shells ?

### DISCUSSION THREAD

**Discussion by markonop93** [ryanoasis/nerd-fonts#1253] [Q&A]
I dont know what I do wrong but I installed few fonts from releases zips with Fontbase then enabled them in cmd, pwsh, Conemu and all 5 charakter material icons are blank square .. how can I display those ?? can you show example

**Comment by markonop93** [comment]
i try both copypaste and code like echo "`u{F1A5E}" in Powershell

**Reply by Finii** [reply]
Which PowerShell version are you using?

The `u{}` syntax has been added with 6.0 (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_special_characters?view=powershell-7.3&viewFallbackFrom=powershell-6#unicode-character-ux)

On my machine it runs flawlessly with PowerShell 7.3.4  ...

I have no clue how or if (ever) it will work with the old `cmd` thing, compare:
https://devblogs.microsoft.com/commandline/windows-command-line-unicode-and-utf-8-output-text-buffer/

But I am no Windows poweruser, other might have better information.
What I guess you need is `chcp 65001` to get any utf support at all.

**Reply by markonop93** [reply]
@Finii powershell 7 last version

**Reply by markonop93** [reply]
i see you use Windows Terminal I try in that now

**Reply by markonop93** [reply]
does not work there too with MesloLGS NF
![image](https://github.com/ryanoasis/nerd-fonts/assets/122524476/62060575-ca2d-4f59-9d2e-fd170b2adb36)

I dont know what i do wrong
I use zips from https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.1

any other debugI can try ?

**Reply by Finii** [reply]
Did you select the correct font in `Windows Terminal`? There is a 'general' and then dedicated detail setting.
I usually have the 'detail' settings on default (i.e. uses the general setting) and change only the general setting.
Maybe show a screenshot of your settings. I can not show right now, because ... having a running Windows box is always work for me :grimacing:

**Reply by markonop93** [reply]
Yes font is correct
everything works only not 5 letter unicodes
![image](https://github.com/ryanoasis/nerd-fonts/assets/122524476/80bd805c-9b73-449e-a72e-6b1eb20e9013)

look
![image](https://github.com/ryanoasis/nerd-fonts/assets/122524476/3a6f940e-7bb5-4e30-b5c7-edde20c3dd35)

**Reply by markonop93** [reply]
it happen with all other fonts I downloaded for example Cascaydia Cove, RobotoMono, Terminess ... is possible that some cache wrong or maybe similar ? if so how I can clear ?

**Reply by Finii** [reply]
I am out of ideas here, but I'm no Windows power user.
Did you notice
* #1246

**Reply by markonop93** [reply]
What is font that works for you @Finii ? Maybe I try that or other one as fresh one

Dont know what you mean by that link . I see no line break

**Reply by Finii** [reply]
I do not think this is a font problem.

Link: Further down... There was a bug in Windows Terminal that has been fixed with 1.18, maybe that is your problem.

**Reply by markonop93** [reply]
Is not
As Im saying problem exist in all other terminals too

I will test any fresh font to see if it cache

**Reply by Finii** [reply]
Some old Windows shell-ish things are broken and fixing them is obviously a) rather hard and b) not high prio, if I read that correctly. I would not expect unicode to work correctly on that (like CMD) any time soon, if at all. As far as I know MS themselves advises Windows Terminal instead of the old things.

Compare the older article that describes the hurdles they (MS) encounter and that I already linked above:
https://devblogs.microsoft.com/commandline/windows-command-line-unicode-and-utf-8-output-text-buffer/

I'm not sure how much that progressed, but I think the old shells are a dead end.
Or that all is fixed now.

Rather strange is that it works on my machines (Windows 10 and 11) but not on yours.

 > I installed few fonts from releases zips with Fontbase

I even do not know what Fontbase is. I would keep it simple:

* uninstall all Nerd Fonts
* reboot
* install one font set from the release zip file via Windows Explorer, select all, install
* try `Libreoffice Writer` or `MS Word` or similar
* try `Windows Terminal`
* Keep in mind that in Windows Terminal the font-to-be-used can be adjusted in several places; reset all to 'default' and just change the 'Defaults' value
* Try `kitty` or some other 3rd party terminal
* This all assumes that no WSL is involved

**Reply by markonop93** [reply]
it work with new download font agave and only in Windows Terminal ... 🎉
![image](https://github.com/ryanoasis/nerd-fonts/assets/122524476/50a72dcc-5a15-498e-82ad-1bd187b7a4f5)

so I think cache issue and very bad support if nothing except Windows Terminal is ok
ConEmu >>>>>> Windows Terminal i guess i skip moved icons until ConEmu fix this

**Reply by markonop93** [reply]
I opened issue there https://github.com/Maximus5/ConEmu/issues/2530

**Comment by Finii** [comment]
Example:

![grafik](https://github.com/ryanoasis/nerd-fonts/assets/16012374/847faaca-7bb3-4aa4-ba26-b60c37a153a3)
